src/ddmd/expression.d

```diff
@@ -5713,6 +5718,15 @@ extern (C++) final class DeclarationExp : Expression
         return new DeclarationExp(loc, declaration.syntaxCopy(null));
     }
 
+    override bool hasCode()
+    {
+        if (auto vd = declaration.isVarDeclaration())
+        {
+            return !(vd.storage_class & (STCmanifest | STCstatic));
```

Comment:
Can you elaborate why the compiler should not complain about static variables? They do not generate code but take up space in the data segment, so a warning seems reasonable nevertheless?

Reply:
Control flow passing by the static variable declaration does not do anything. Your argument justifies warnings for unused static variables, but not static variables that control flow does not reach.

Reply:
@tgehr I agree.

Reply:
OK, but is there actually a case where a _static variable declaration_ is not reached by control flow __and__ the variable is still _accessible from some place reached by code flow_?

```d
void main()
{
    return;

    {
        static int x;
    }
}
```
My line of thought goes like this: If the declaration is unreachable, all code in the scope of the declaration following the declaration is unreachable as well. The variable is only accessible in the same scope (or a nested scope) defined after the variable declaration. So, such a variable seems to be always unused/used only by dead code.

OTOH `int x = void;` also does not generate any code, so I don't really understand the `passing by the static variable declaration does not do anything` point. Or are you just arguing that these cases are still invalid, but should produce a `variable unused` message instead of the `statement not reachable` message?